dockerrotate/containers.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime, timedelta
+from dateutil import parser
+from dateutil.tz import tzutc
+import re
+
+from docker.errors import APIError
+
+from dockerrotate.filter import include_image
+
+
+TIME_REGEX = re.compile(r'((?P<days>\d+?)d)?((?P<hours>\d+?)h)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')  # noqa
+
+
+def parse_time(time_str):
+    """
+    Parse a human readable time delta string.
+    """
+    parts = TIME_REGEX.match(time_str)
+    if not parts:
+        raise Exception("Invalid time delta format '{}'".format(time_str))
+    parts = parts.groupdict()
+    time_params = {}
+    for (name, param) in parts.iteritems():
+        if param:
+            time_params[name] = int(param)
+    return timedelta(**time_params)
+
+
+def include_container(container, args):
+    """
+    Return truthy if container should be removed.
+    """
```

Comment:
Only exited containers or containers that have been created but not started are considered for removal.

Note that I removed the support for data containers.
In latest docker versions data volumes are first class objects and should be created with `docker volume` command instead of creating data containers.